staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -431,14 +435,7 @@ func (s *store) conditionalDelete(
 		}
 
 		if !skipTransformDecode {
-			if len(txnResp.Responses) == 0 || txnResp.Responses[0].GetResponseDeleteRange() == nil {
-				return errors.New(fmt.Sprintf("invalid DeleteRange response: %v", txnResp.Responses))
-			}
-			deleteResp := txnResp.Responses[0].GetResponseDeleteRange()
-			if deleteResp.Header == nil {
-				return errors.New("invalid DeleteRange response - nil header")
-			}
-			err = decode(s.codec, s.versioner, origState.data, out, deleteResp.Header.Revision, clusterName)
+			err = decode(s.codec, s.versioner, origState.data, out, txnResp.Revision, clusterName, shardName)
```

Comment:
Do we remember why this is different from upstream?
@sttts maybe you recall?

Reply:
Upstream's `conditionalDelete` was the same in v1.31.0 too: https://github.com/kubernetes/kubernetes/blob/v1.31.0/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L349-L385

Now, instead of explicitly doing `Txn-If-Then-Else-Commit`, there is a dedicated [`OptimisticDelete`](https://github.com/kubernetes/kubernetes/blob/v1.32.0/vendor/go.etcd.io/etcd/client/v3/kubernetes/client.go#L109-L128)`, and so `deleteResp` is not used anymore. Is this what you meant, @mjudeikis